File: examples/python/company_profile.py
"""Resolve a UK company by name, then fetch its full profile (flagship call).

Usage: ORO_API_KEY=oro_... python company_profile.py "Serco"
Requires: pip install httpx
"""

import os
import sys

import httpx

API_KEY = os.environ["ORO_API_KEY"]  # https://app.oro-intel.com/dashboard/developers
BASE = "https://api.oro-intel.com/v1"
HEADERS = {"Authorization": f"Bearer {API_KEY}"}

query = sys.argv[1] if len(sys.argv) > 1 else "Serco"

with httpx.Client(base_url=BASE, headers=HEADERS, timeout=30) as client:
    # 1. Resolve name -> company_number (5 credits)
    # Returns the single best match: core record incl. companies_house_number
    search = client.get("/companies/search", params={"name": query}).raise_for_status().json()
    number = search["companies_house_number"]
    if not number:
        sys.exit(f"Best match {search['name']!r} has no Companies House number — try a more specific name.")
    print(f"Resolved {query!r} -> {number} (credits_charged={search['credits_charged']})")

    # 2. Full profile: core record + every contract won, one call (12 credits)
    profile = client.get(f"/companies/{number}/profile").raise_for_status().json()
    print("name:", profile["core"].get("name"))
    print("contracts won:", profile["contracts"]["contract_count"])
    print("credits_charged:", profile["credits_charged"])
    print("credits_remaining:", profile["credits_remaining"])

File: examples/javascript/company-profile.mjs
// Resolve a UK company by name, then fetch its full profile (flagship call).
// Usage: ORO_API_KEY=oro_... node company-profile.mjs "Serco"
// Requires Node 18+ (built-in fetch). No dependencies.

const API_KEY = process.env.ORO_API_KEY; // https://app.oro-intel.com/dashboard/developers
if (!API_KEY) throw new Error("Set ORO_API_KEY");
const BASE = "https://api.oro-intel.com/v1";
const headers = { Authorization: `Bearer ${API_KEY}` };
const query = process.argv[2] ?? "Serco";

async function get(path) {
  const res = await fetch(`${BASE}${path}`, { headers });
  if (!res.ok) throw new Error(`${res.status} ${await res.text()}`);
  return res.json();
}

// 1. Resolve name -> company_number (5 credits)
const search = await get(`/companies/search?name=${encodeURIComponent(query)}`);
const number = search.items[0].company_number;
console.log(`Resolved "${query}" -> ${number} (credits_charged=${search.credits_charged})`);

// 2. Full profile: core record + every contract won, one call (12 credits)
const profile = await get(`/companies/${number}/profile`);
console.log("name:", profile.core?.name);
console.log("contracts won:", profile.contracts?.contract_count);
console.log("credits_charged:", profile.credits_charged);
console.log("credits_remaining:", profile.credits_remaining);
